mongo/integration/gridfs_test.go

```diff
@@ -369,6 +370,34 @@ func TestGridFS(x *testing.T) {
 			_, err = bucket.OpenDownloadStream(oid)
 			assert.Equal(mt, gridfs.ErrMissingChunkSize, err, "expected error %v, got %v", gridfs.ErrMissingChunkSize, err)
 		})
+		mt.Run("cursor error during read after downloading", func(mt *mtest.T) {
+			// To simulate a cursor error we upload a file larger than the 16MB default batch size,
+			// so Read() will actually execute a getMore against the server. Since the ReadDeadline is
+			// set in the past, this should cause a timeout.
+
+			fileName := "read-error-test"
+			fileData := make([]byte, 17000000)
+
+			bucket, err := gridfs.NewBucket(mt.DB)
+			assert.Nil(mt, err, "NewBucket error: %v", err)
+			defer func() { _ = bucket.Drop() }()
+
+			dataReader := bytes.NewReader(fileData)
+			_, err = bucket.UploadFromStream(fileName, dataReader)
+			assert.Nil(mt, err, "UploadFromStream error: %v", err)
+
+			ds, err := bucket.OpenDownloadStreamByName(fileName)
+			assert.Nil(mt, err, "OpenDownloadStreamByName error: %v", err)
+
+			err = ds.SetReadDeadline(time.Now().Add(-1 * time.Second))
+			assert.Nil(mt, err, "SetReadDeadline error: %v", err)
+
+			p := make([]byte, len(fileData))
+			_, err = ds.Read(p)
```

Comment:
Can you parameterize this test case, or add a test for `Skip` as well? This reminded me about GODRIVER-1806, and how `Read` and `Skip` are near duplicates, but `Skip` avoids extra copies.

Reply:
Ah great catch. `Read` and `Skip` both call `fillBuffer`, where the bug in error surfacing was. Added a different test case for `Skip` (wasn't sure how to parameterize given `Skip` and `Read` have different function signatures).